first_hour_on_vwb/widget_utils.py

```diff
@@ -57,6 +57,26 @@ def list_group_members(json_string):
     html += "</table>"
     return html
 
+def list_workspace_ids(json_string):
+    """ Return a list of workspace IDs.
+    """
+    json_data = json.loads(json_string)
+    id_list = []
+    for row in json_data:
+        ws_id = row['id']
+        ws_properties = row['properties']
+        if 'terra-type' in ws_properties:
+            terra_type = ws_properties['terra-type']
+            if terra_type == 'data-collection':
+                continue
+            # Workspaces do not have an explictly set terra-type
+            # but may have user-set properties.
+            else:
+                id_list.append(ws_id)
+        else:
+            id_list.append(ws_id)
+    return id_list
```

Comment:
This may be cleaner:
```suggestion
    for row in json_data:
        ws_id = row['id']
        ws_properties = row['properties']
        if ws_properties.get('terra-type') == 'data-collection':
            # Workspaces do not have an explictly set terra-type
            # but may have user-set properties.
            continue
        id_list.append(ws_id)
    return id_list
```

Reply:
done!